### PR TITLE
Heidenhain ND287 Position Display Unit Driver

### PR DESCRIPTION
**Pull request by samcondon4**
This PR implements a basic driver for the Heidenhain ND287 Position Display Unit. This driver is pretty basic, though I would like to hear feedback on the `id` and `error` properties. Both of the queries that these properties trigger return strings with several lines of characters. That is, the instrument returns several carriage returns and line feeds in the return string. By just creating a normal property via `Instrument.measurement`, the read operation is terminated on the first carriage return, so the whole string is not retrieved. To account for this I used the low-level `adapter.connection.write()` and `adapter.connection.read_bytes()` methods, though I am not sure that this is the cleanest implementation. Do any of the reviewers have suggestions of better ways to implement these properties? Perhaps a few additions to the `Instrument` base class are in order to accomplish this kind of query in a cleaner way?

Thanks in advance!

### CODE REVIEW

File: docs/api/instruments/heidenhain/index.rst
Comment (bilderbuchi) [inline docs/api/instruments/heidenhain/index.rst:2]: this warning above means that you need to add a reference in  docs/api/instruments/index.rst

File: pymeasure/instruments/heidenhain/nd287.py
Comment (bilderbuchi) [inline pymeasure/instruments/heidenhain/nd287.py:30]: You can call it ND287

File: pymeasure/instruments/heidenhain/nd287.py
Comment (bilderbuchi) [inline pymeasure/instruments/heidenhain/nd287.py:59]: please move the init to the top of the class

File: pymeasure/instruments/heidenhain/nd287.py
Comment (bilderbuchi) [inline pymeasure/instruments/heidenhain/nd287.py:53]: Have you tested that this works correctly, i.e. you can switch units after creating an instance by using self.units?

Also, be aware of the caveat that, if you have more than one ND287 at the same time, I think they then can't have different settings. A proper fix of this will be (WIP, but slowly), see #250.

File: pymeasure/instruments/heidenhain/nd287.py
Comment (bilderbuchi) [inline pymeasure/instruments/heidenhain/nd287.py:80]: do you need this property? Is it not better to pull this into the `check_errors` method?

File: pymeasure/instruments/heidenhain/nd287.py
Comment (bilderbuchi) [inline pymeasure/instruments/heidenhain/nd287.py:118]: v there is an empty file  pymeasure/instruments/nd287.py shown below - please delete.

File: pymeasure/instruments/heidenhain/nd287.py
Comment (bilderbuchi) [inline pymeasure/instruments/heidenhain/nd287.py:28]:
```suggestion
import logging
from pyvisa.errors import VisaIOError
from pymeasure.instruments import Instrument

log = logging.getLogger(__name__)
log.addHandler(logging.NullHandler())
```
then use `log` instead of `logging` below.

**Review by bilderbuchi — CHANGES_REQUESTED** [review]
Looking good, some small things, really.

File: docs/api/instruments/heidenhain/index.rst
Comment (samcondon4) [reply docs/api/instruments/heidenhain/index.rst:2]: Oops. Fixed that.

File: pymeasure/instruments/heidenhain/nd287.py
Comment (samcondon4) [reply pymeasure/instruments/heidenhain/nd287.py:59]: Got it.

File: pymeasure/instruments/heidenhain/nd287.py
Comment (samcondon4) [reply pymeasure/instruments/heidenhain/nd287.py]: I did test this and it worked (at least for one device). I believe that you are correct though that multiple ND287 devices could not have different settings. To address this, I've implemented the `position` property manually so that it can query the instance `_units` attribute to apply the appropriate scale factor. I will keep an eye on #250 though and perhaps this property can be updated when #250 is accepted.

File: pymeasure/instruments/heidenhain/nd287.py
Comment (samcondon4) [reply pymeasure/instruments/heidenhain/nd287.py]: Nope, I think it is better to pull it into `check_errors`. The latest commit addresses that.

File: pymeasure/instruments/heidenhain/nd287.py
Comment (samcondon4) [reply pymeasure/instruments/heidenhain/nd287.py:107]: Not sure how I did that. It has been removed.

File: pymeasure/instruments/heidenhain/nd287.py
Comment (samcondon4) [reply pymeasure/instruments/heidenhain/nd287.py:28]: Got it.

File: pymeasure/instruments/heidenhain/nd287.py
Comment (bilderbuchi) [reply pymeasure/instruments/heidenhain/nd287.py]: A solution for #250 is nearing completion now, so I'd wait until #376 is merged, and then use the new functionality here.

Comment (bilderbuchi) [reply pymeasure/instruments/heidenhain/nd287.py]: In the meantime, we have activated a linter, so there are a couple of linting reports in the "Files changed" tab, which you could fix when you have an opportunity.

Comment (bilderbuchi) [reply pymeasure/instruments/heidenhain/nd287.py]: Great to hear! A couple of linter issues have reappeared -- you can run flake8 locally (`flake8 .`) to check for issues (if your editor does not do that automatically)

Comment (samcondon4) [reply pymeasure/instruments/heidenhain/nd287.py]: Do you still see the linter issues on your end? I believe my latest commit has addressed them. I ran flake8 locally and don't see any issues in the remote or local checks on my end anymore.

Comment (bilderbuchi) [reply pymeasure/instruments/heidenhain/nd287.py]: Hm, I guess some UI state in the browser must have been stale? Anyway, nothing to see anymore, thanks!

**Pull request merged**
